TCF4.py

```diff
@@ -0,0 +1,121 @@
+from Bio import SeqIO
+from Bio.SeqUtils import GC
+
+def filter_fastq(input_path: str, quality_threshold: int, output_filename="final_filtered.fastq",  gc_bounds=(40, 60), length_bounds=(50, 350)):
+    filename = input_path
+    records = SeqIO.parse(filename, "fastq")
+    ###quality filter
+    good_reads = (rec for rec in records if min(rec.letter_annotations["phred_quality"]) >= quality_threshold)
+    result_quality = SeqIO.write(good_reads, "good_quality.fastq", "fastq")
+    result_quality_GC = SeqIO.parse("good_quality.fastq", "fastq")
+    ###GC content filter
+    min_gc_content = gc_bounds[0]
+    max_gc_content = gc_bounds[1]
```

Comment:
`gc_bounds` , по условию задания, может, в том числе, принимать только верхний порог в виде `int`. В этом случае у тебя здесь упадёт код.

```suggestion
    if isinstance(gc_bounds, int):
        gc_bounds = (0, gc_bounds)
    min_gc_content, max_gc_content = gc_bounds
```